generated/language/src/main/java/com/google/cloud/language/v1/spring/LanguageServiceSpringAutoConfiguration.java

```diff
@@ -0,0 +1,471 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.language.v1.spring;
+
+import com.google.api.core.BetaApi;
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.api.gax.core.ExecutorProvider;
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.rpc.HeaderProvider;
+import com.google.api.gax.rpc.TransportChannelProvider;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.LanguageServiceSettings;
+import com.google.cloud.spring.core.DefaultCredentialsProvider;
+import com.google.cloud.spring.global.GlobalProperties;
+import java.io.IOException;
+import java.util.Collections;
+import javax.annotation.Generated;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+
+// AUTO-GENERATED DOCUMENTATION AND CLASS.
+/**
+ * Auto-configuration for {@link LanguageServiceClient}.
+ *
+ * <p>Provides auto-configuration for Spring Boot
+ *
+ * <p>The default instance has everything set to sensible defaults:
+ *
+ * <ul>
+ *   <li>The default transport provider is used.
+ *   <li>Credentials are acquired automatically through Application Default Credentials.
+ *   <li>Retries are configured for idempotent methods but not for non-idempotent methods.
+ * </ul>
+ */
+@Generated("by gapic-generator-java")
+@BetaApi("Autogenerated Spring autoconfiguration is not yet stable")
+@AutoConfiguration
+@ConditionalOnClass(LanguageServiceClient.class)
+@ConditionalOnProperty(
+    value = "com.google.cloud.language.v1.spring.auto.language-service.enabled",
+    matchIfMissing = false)
+@EnableConfigurationProperties({LanguageServiceSpringProperties.class, GlobalProperties.class})
+public class LanguageServiceSpringAutoConfiguration {
+  private final LanguageServiceSpringProperties clientProperties;
+  private final GlobalProperties globalProperties;
```

Comment:
What are these global properties? I can't find this class.

Reply:
If we are going to have a handwritten shared class, it makes more sense for us to move away from the using the folder name `generated`.